### DISCUSSION TITLE
How to rename?

### DISCUSSION THREAD

**Discussion by 9mm** [mrcjkb/rustaceanvim#448] [Q&A] [answered]
### Feature description

I was shocked to find out my friends vscode can rename variables and traits across a project... i checked rustacean to see if this existed months ago and didnt see anything, but i never actually checked rust analyzer docs. low and behold it can do this. so many 1000's of hours wasted 💀

how do i use this from within rustacean? https://rust-analyzer.github.io/manual.html#rename

**Comment by mrcjkb** [comment, marked as answer]
Tha's a built-in Neovim feature. See `:h lsp` and `:h vim.lsp.buf.rename`.

**Reply by 9mm** [reply]
fuuuuuuuuu........ i see why i didnt have that now. i had another rename that was localized to the file which was overwriting rename symbol. I thought i was using buf.rename the whole time...
